src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs

```diff
@@ -274,6 +274,40 @@ public static string GetTempPath(string workingDir)
             }
         }
 
+        private static string FindDotNetExecutable()
+        {
+            var expectedPath = Environment.GetEnvironmentVariable("DOTNET_HOST_PATH");
+            if (!string.IsNullOrEmpty(expectedPath))
+            {
+                return expectedPath;
+            }
+
+#if NET
+            expectedPath = System.Environment.ProcessPath;
+#else
+            expectedPath = Process.GetCurrentProcess().MainModule.FileName;
+#endif
+
+            if ("dotnet".Equals(Path.GetFileNameWithoutExtension(expectedPath)))
```

Comment:
Generally we recommend to always explicitly pass comparer to string comparisons so if nothing else it's clear to the reader.